gnomad_qc/v5/sample_qc/relatedness.py

```diff
@@ -0,0 +1,180 @@
+    accelerator_count: int = 1,
+    accelerator_type: str = "nvidia-tesla-a100",
+    requester_pays_project: str = "terra-vpc-sc-93ccd8d2",
+) -> None:
+    """
+    Print the command to submit a dsub job for running cuKING.
+
+    :param cuking_input_path: Path to the cuKING input Parquet files.
+    :param cuking_output_path: Path to the cuKING output Parquet files.
+    :param min_emission_kinship: Minimum kinship threshold for emitting a pair of
+        samples in the relatedness output.
+    :param cuking_split_factor: Split factor to use for splitting the full relatedness
+        matrix table into equally sized submatrices that are computed independently to
+        parallelize the relatedness computation and decrease the memory requirements.
+        For example, to halve memory requirements, the full matrix can be split into
+        equally sized submatrices (i.e. a 'split factor' of 4). Only the
+        'upper triangular' submatrices need to be evaluated due to the symmetry of the
+        relatedness matrix, leading to 10 shards. Default is 4.
+    :param location: Location to run the dsub job. Default is "us-central1".
+    :param machine_type: Machine type to use for the dsub job. Default is "a2-highgpu-1g".
+    :param accelerator_count: Number of accelerators to use for the dsub job. Default is 1.
+    :param accelerator_type: Type of accelerator to use for the dsub job.
+        Default is "nvidia-tesla-a100".
+    :param requester_pays_project: Project ID to use for requester pays buckets.
+        Default is "terra-vpc-sc-93ccd8d2".
+    :return:
+    """
+    logger.info(
+        "Printing a command that can be used to submit a dsub job to run "
+        "cuKING on the files created by --prepare-cuking-inputs."
+    )
+    logger.warning(
+        "This printed command assumes that the cuKING directory is in the same "
+        "location where the command is being run and that $PROJECT_ID is set!"
+    )
+    SPLIT_FACTOR = cuking_split_factor
+    TOTAL_SHARDS = SPLIT_FACTOR * (SPLIT_FACTOR + 1) / 2
+    print(
+        textwrap.dedent(
+            f"""\
+            for SHARD_INDEX in $(seq 0 $(({TOTAL_SHARDS} - 1))); do \\
+                cuKING_dsub \\
+                --location={location} \\
+                --machine-type={machine_type} \\
+                --accelerator-count={accelerator_count} \\
+                --accelerator-type={accelerator_type} \\
+                --command="cuking \\
+                --input_uri={cuking_input_path} \\
+                --output_uri={cuking_output_path} \\
+                --requester_pays_project={requester_pays_project} \\
+                --kin_threshold={min_emission_kinship} \\
+                --split-factor={SPLIT_FACTOR} \\
+                --shard-index=SHARD_INDEX" \\
+            done
+            """
+        )
+    )
+
+
+def main(args):
+    """Compute relatedness estimates among pairs of samples in the callset."""
+    # test = args.test
```

Comment:
Assuming these will be used later? Did you want to add a test?

Reply:
I added these initially with code to run the QC MT > parquet conversion but ran into issues with that (will respond in more detail in the other thread)